testing/file_test/autoupdate.cpp

```diff
@@ -71,25 +73,31 @@ auto FileTestAutoupdater::CheckLine::RemapLineNumbers(
       return;
     }
 
-    // Update the cursor offset from the match.
-    line_offset = matched_line_number.begin() - line_.c_str();
+    // Map the matched filename to its file number.
+    auto matched_file_number = file_number();
+    if (replacement_->has_file) {
+      auto it = file_to_number_map.find(matched_filename);
+      if (it != file_to_number_map.end()) {
+        matched_file_number = it->second;
+      }
+    }
```

Comment:
I'm trying to figure out why this is setting `matched_file_number` instead of just dropping it on the floor... Were you seeing that the `new_line_number` remapping done in lines 89-97 need to be applied also to other files?